### PR TITLE
fix: resolved eslint issues

### PR DESCRIPTION
**Pull request by nikhilgarg26**
# Description

Updated @typescript-eslint/eslint-plugin and @typescript-eslint/parser to the latest version.

## Which issue(s) does this PR relate to?

#995 

## Testing

Storybook version also needs to be updated and the defined tests used to resolve errors.
Total of 6 errors got.

## Checklist

- [ ] This PR is associated with a JIRA and it is mentioned in the commit message footer ("Closes …")
- [ ] Significant changes have been tested downstream to avoid breaking changes
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [ ] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -98,8 +99,8 @@
     "@types/react": "^18.2.21",
     "@types/react-dom": "^18.2.7",
     "@types/react-virtualized": "^9.21.22",
-    "@typescript-eslint/eslint-plugin": "^5.59.11",
-    "@typescript-eslint/parser": "^5.59.11",
+    "@typescript-eslint/eslint-plugin": "^6.7.2",
+    "@typescript-eslint/parser": "^6.7.2",
```
Comment (nataliepina) [inline package.json:103]: We only want these changes to address updating these two:
```
"@typescript-eslint/eslint-plugin": "^6.7.2",
"@typescript-eslint/parser": "^6.7.2",
```

Can you please remove the other updates from this file so we can isolate the changes here and better test what has been updated. Thank you!

File: commitlint.config.cjs [hunk 2]
```diff
@@ -1,3 +1,4 @@
-module.exports = {
-  extends: ["@commitlint/config-conventional"]
-};
+
+
+module.exports = { extends: ['@commitlint/config-conventional'] };
```
Comment (nataliepina) [inline commitlint.config.cjs:3]: Looks like this is just a linting change can you please revert it?
I don't think we need the single backtick change in here.

**Comment by nikhilgarg26** [comment]
If i just install the necessary updations and not install rest it gives error 
image
on `git push origin nikhil/build/update-dev-dependencies`  as the husky performs pre-push tests `npm run test`
What can be the suggested solution to it ?

**Comment by clintonmedbery** [comment]
Hey @nikhilgarg26 thanks for your contribution. I don't think you should need all the other updates or a new babel file. I made a few of these updates and the tests passed when running `npm run test`. What kind of errors are you seeing when trying to push? What do the tests say? What version of node/npm are you running?

**Comment by nikhilgarg26** [comment]
@clintonmedbery i am using Node v18.16.0 and npm v9.5.1 when i try to push the commits by `git push origin nikhil/build/update-dev-dependencies` it starts performing tests which fails giving 
image
and eventually all tests are FAILED
image

**Comment by clintonmedbery** [comment]
Do you get the same thing running `npm run test`? Regardless I would remove all those additional dependency updates and work from there. I can get tests to pass without that babel file etc.

**Comment by nikhilgarg26** [comment]
Yes i get the same thing running `npm run test` even the git commit was not working commited my files using `git commit -m "***" --no-verify`
steps after cloning i followed as given in [contribution.md](https://github.com/d2iq/ui-kit/wiki/How-to-Contribute#contribution-guide)
- npm i
- npm run storybook
- npm install @typescript-eslint/parser@latest
- npm install @typescript-eslint/eslint-plugin@latest
- some warning removed by npm run lint -- --fix
- some errors occured due to updation of eslint
- npm run test:types

**Comment by nataliepina** [comment]
> Yes i get the same thing running `npm run test` even the git commit was not working, commited my files using `git commit -m "***" --no-verify` steps after cloning i followed as given in [contribution.md](https://github.com/d2iq/ui-kit/wiki/How-to-Contribute#contribution-guide)

> * npm i
> * npm run storybook
> * npm install @typescript-eslint/parser@latest
> * npm install @typescript-eslint/eslint-plugin@latest
> * some warning removed by npm run lint -- --fix
> * some errors occured due to updation of eslint
> * npm run test:types

I've followed these steps myself here that you've outlined and If I go through all of those steps and I run `npm run test` I see all tests passing. If I run `npm run test:types` I see the 6 issues that need to be resolved individually. You'll have to manually resolve these. `npm run lint -- --fix` will not fix them. 
I don't see the failures that you are encountering with the 75 tests, that's why it seems there is a configuration issue. 
Are you sure you've removed all of the other changes it and only made the updates to the 2 depdencies when you're encountering that? You may need to remove node_modules and try to npm install again if you've removed the other items.
We shouldn't need the updates to the other dependencies or need a babelrc file created with this work.  
It might be a good idea to start from the top and make sure you're able to run the repository and tests successfully prior to any changes.

File: package.json [hunk 3]
```diff
@@ -123,9 +124,11 @@
     "jest-environment-jsdom": "^29.6.4",
     "jest-runner-prettier": "^1.0.0",
     "prettier": "^3.0.3",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
```
Comment (nataliepina) [inline package.json:128]: We definitely do not want this included in here.
We use peerDependcies for React in this project.

```
 "peerDependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  },
```

**Pull request closed without merging**